### DISCUSSION TITLE
Monaco Editor Wrapper for Multiple Languages

### DISCUSSION THREAD

**Discussion by Plochem** [eclipse-langium/langium#1460] [Q&A]
Is there an example on using the Monaco Editor Wrapper for multiple dependent languages? In the editor, I want to be able to create multiple "tabs"/models where I can set the language for each and they can "see" each other. I used the yeoman generator for langium and defined my languages just like the ones in multiple languages guide. 

I've tried to use the extended wrapper config which allowed me to set multiple languages and grammars. The editor works fine only for whichever language I set for the languageId. Obviously, the parts that depend on the other language wouldn't work. I tried calling createModel & setModel from the Monaco API. But whenever I type in the editor, all the syntax highlighting and validation stop working. I also get this error whenever I type anything into the editor: 
<img width="718" alt="Screenshot 2024-04-18 at 2 08 39 PM" src="https://github.com/eclipse-langium/langium/assets/30739927/33111753-8555-4ae4-afb1-f0c03b328c07">

**Comment by kaisalmen** [comment]
Hi @Plochem please open an issue in the corresponding [repository](https://github.com/TypeFox/monaco-languageclient). There is already an issue regarding the "inflexibility" or enforcement of `languageId`, see [#639](https://github.com/TypeFox/monaco-languageclient/issues/639). I am actively working on improving that. Either add your problem to the existing issue one or open a new issue or discussion. Thank you.
